.github/workflows/build.yml

```diff
@@ -0,0 +1,50 @@
+# Copyright The Ratify Authors.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: build
+
+on:
+  push:
+    branches: 
+      - main
+      - release-*
+  pull_request:
+    branches:
+      - main
+      - release-*
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      checks: write
+    strategy:
+      matrix:
+        go-version: ['1.22', '1.23']
+      fail-fast: true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Go ${{ matrix.go-version }} environment
+        uses: actions/setup-go@v5
```

Comment:
in ratify repo, we usually pin to a specific version and add tag as comment.
Like: actions/setup-go@3041bf56c941b39c61721a86cd11f3bb1338122a # v5.2.0

Reply:
Yepp, this will be the next step.